maven-core/src/main/java/org/apache/maven/plugin/version/internal/DefaultPluginVersionResolver.java

```diff
@@ -384,6 +401,36 @@ private PluginVersionResult resolveFromProject( PluginVersionRequest request, Li
         return null;
     }
 
+    @SuppressWarnings( "unchecked" )
+    private Map<String, PluginVersionResult> getCache( SessionData data )
+    {
+        Map<String, PluginVersionResult> cache = ( Map<String, PluginVersionResult> ) data.get( CACHE_KEY );
+        while ( cache == null )
```

Comment:
Why not use `computeIfAbsent?

Reply:
That would be awesome, but `SessionData` interface is not a `Map` unfortunately !...

Reply:
The problem is that this spot is not atomic,  is it?